discovery/config.go

```diff
@@ -0,0 +1,56 @@
+package discovery
+
+import "crypto/tls"
+
+type Config struct {
+	// Addresses is a DNS name or exec command for go-netaddrs.
+	Addresses string
+	// GRPCPort is the gRPC port to connect to. This must be the
+	// same for all Consul servers for now. Defaults to 8502.
+	GRPCPort int
+
+	// ServerWatchDisabled disables opening the ServerWatch gRPC
+	// stream. This should be used when your Consul servers are
+	// behind a load balancer, for example, since the server addresses
+	// returned in the ServerWatch stream will differ from the load
+	// balancer address.
+	ServerWatchDisabled bool
+
+	TLS         *tls.Config
```

Comment:
Opted against having this library read from the filesytem. So we'll load certs in consul-dataplane into a tls.Config, which we can plug into the grpc connection easily.

Reply:
+1